jerry-core/ecma/base/ecma-globals.h

```diff
@@ -227,56 +306,61 @@ typedef struct
 } ecma_getter_setter_pointers_t;
 
 /**
- * Description of ecma-property
+ * Property data.
  */
-typedef struct ecma_property_t
+typedef union
 {
-  /** Compressed pointer to next property */
-  mem_cpointer_t next_property_p;
+  ecma_value_t value; /**< value of a property */
+  ecma_getter_setter_pointers_t getter_setter_pair; /**< getter setter pair */
+} ecma_property_value_t;
 
-  /** Property's flags (ecma_property_flags_t) */
-  uint8_t flags;
+/**
+ * Property pair.
+ */
+typedef struct
+{
+  ecma_property_header_t header; /**< header of the property */
```

Comment:
I'm not sure why this needs to be in a separate structure? Cannot we have the contents of `ecma_property_header_t` right here?